pinot-core/src/main/java/org/apache/pinot/core/indexsegment/IndexSegment.java

```diff
@@ -29,6 +29,15 @@
 @InterfaceAudience.Private
 public interface IndexSegment {
 
+  public enum IndexSegmentType{
```

Comment:
Why do we need this? On this level we should not have concept of IMMUTABLE or MUTABLE. We can do `if (indexSegment instanceOf MutableSegment)` if necessary.

Reply:
I don't think I agree with this. Exposing a type is a much cleaner way of handling this than doing instanceOf.